docs/beyond_basics/101-182-catalog.rst

```diff
@@ -3,7 +3,573 @@
+  For an object of type ``dataset``, this will be the ``dataset_id`` and ``dataset_version``
+  of the actual dataset. For an object of type ``file``, this will be the ``dataset_id``
+  and ``dataset_version`` of its parent dataset (i.e. the dataset which the file forms part of).
+- Each metadata object of type ``file`` should have a ``path`` key for which the value
+  specifies exactly where the file is located relative to the root directory of its parent dataset.
+- Datasets can have subdatasets.
+
+The Catalog schema specifies exactly which fields are required and which data types
+are accepted for each key/value-pair. For an improved understanding of the Catalog schema,
+you can inspect the `JSON documents here <https://github.com/datalad/datalad-catalog/tree/main/datalad_catalog/schema>`_ (``jsonschema_*``).
+
+Sample metadata
+"""""""""""""""
+
+Let's look at a toy example of metadata that adheres to the Catalog schema.
+
+First a dataset:
+
+.. code-block::
+
+   {
+       "type": "dataset",
+       "dataset_id":"abcd",
+       "dataset_version":"1234",
+       "name": "My toy dataset",
+       "short_name": "My toy dataset",
+       "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec justo tellus. Nunc sagittis eleifend magna, eu blandit arcu tincidunt eu. Mauris pharetra justo nec volutpat euismod. Curabitur bibendum vitae nunc a pharetra. Donec non rhoncus risus, ac consequat purus. Pellentesque ultricies ut enim non luctus. Sed viverra dolor enim, sed blandit lorem interdum sit amet. Aenean tincidunt et dolor sit amet tincidunt. Vivamus in sollicitudin ligula. Curabitur volutpat sapien erat, eget consectetur mauris dapibus a. Phasellus fringilla justo ligula, et fringilla tortor ullamcorper id. Praesent tristique lacus purus, eu convallis quam vestibulum eget. Donec ullamcorper mi neque, vel tincidunt augue porttitor vel.",
+       "doi": "",
+       "url": ["https://github.com/jsheunis/multi-echo-super"],
+       "license": {
+         "name": "CC BY 4.0",
+         "url": "https://creativecommons.org/licenses/by/4.0/"
+       },
+       "authors": [
+           {
+               "givenName":"Stephan",
+               "familyName":"Heunis",
+           }
+       ],
+       "keywords": ["lorum", "ipsum", "foxes"],
+       "funding": [
+           {
+               "name":"Stephans Bank Account",
+               "identifier":"No. 42",
+               "description":"Nothing to see here"
+           }
+       ],
+       "extractors_used": [
+           {
+               "extractor_name": "stephan_manual",
+               "extractor_version": "1",
+               "extraction_parameter": {},
+               "extraction_time": 1652340647.0,
+               "agent_name": "Stephan Heunis",
+               "agent_email": ""
+           }
+       ]
+   }
+
+And then two files of the dataset:
+
+ .. code-block::
+
+   {
+      "type": "file"
+      "dataset_id": "abcd",
+      "dataset_version": "1234",
```

Comment:
There might be value in using an ID and version that are structurally identical to the real things (a UUID and a Git SHA). Doesn't add much weight, and would match the actual appearance of a real catalog better. Maybe?